src/rgw/rgw_bucket.cc

```diff
@@ -635,6 +635,43 @@ int RGWBucket::init(rgw::sal::RGWRadosStore *storage, RGWBucketAdminOpState& op_
   return 0;
 }
 
+bool RGWBucket::rgw_find_bucket_by_id(CephContext *cct, RGWMetadataManager *mgr,
+                           const string& marker, const string& bucket_id, rgw_bucket* bucket_out)
```

Comment:
looks good, but it can just be a free function outside of `class RGWBucket`

Reply:
Thanks @cbodley. Fixed!